table/block.h

```diff
@@ -199,6 +205,8 @@ class Block {
   // the encoded value (kDisableGlobalSequenceNumber means disabled)
   const SequenceNumber global_seqno_;
 
+  DataBlockHashIndex data_block_hash_index_;
```

Comment:
DataBlockHashIndex is too big to have it in every class Block. Can we reduce it as much as possible? Can we have two unit32_t, one for offset of the starting, and one for number of buckets?

Reply:
I think we can improve further to only one uint32 by only storing the num_bucket. maps_start can be calculated from num_restarts. Let me try.

Reply:
Just pushed a commit, which shrinks the size from
`3 * sizeof(char*) + 2 * sizeof(uint16_t)` to `sizeof(uint16_t)`